pydicom/filereader.py

```diff
@@ -230,6 +233,15 @@ def data_element_generator(fp,
     element_struct_unpack = element_struct.unpack
     defer_size = size_in_bytes(defer_size)
 
+    tag_set = None
+    if specific_tags is not None:
+        tag_set = set()
```

Comment:
Ah well, that's ok then - I was more concerned here about the cost of union, but in the end its all irrelevant compared with the I/O costs. Convinced me :)